packages/ui-components/src/components/radio-button/radio-button.tsx

```diff
@@ -0,0 +1,66 @@
+import { Component, Host, h, Prop, EventEmitter, Event, State, Watch } from '@stencil/core';
+import { throttle } from 'lodash-es';
+
+@Component({
+	tag: 'kv-radio-button',
+	styleUrl: 'radio-button.scss',
+	shadow: true
+})
+export class KvRadioButton {
+	/** (optional) Sets the button as checked when initializing */
+	@Prop({ reflect: true }) checked?: boolean = false;
+	/** (optional) Sets this button styling to be disabled and disables click events */
+	@Prop({ reflect: true }) disabled?: boolean = false;
+	/** (optional) Adds a label aside the button */
+	@Prop({ reflect: true }) label?: string = '';
+
+	/** Internal checked / unchecked state */
+	@State() isChecked: boolean = this.checked;
+	/** Internal enabled / disabled state */
+	@State() isDisabled: boolean = this.disabled;
+
+	/** Emits when there's a change in state internally */
+	@Event() checkedChange: EventEmitter<boolean>;
+
+	/** Watch the `checked` property and update internal state accordingly */
+	@Watch('checked')
```

Comment:
`Watch`'es should be next the respective `prop`. because if you change one you need change the other